README.md

```diff
@@ -42,44 +42,52 @@ from tidb_vector.sqlalchemy import VectorType
 engine = create_engine('mysql://****.root:******@gateway01.xxxxxx.shared.aws.tidbcloud.com:4000/test')
 Base = declarative_base()
 
-class Test(Base):
-    __tablename__ = 'test'
+class Document(Base):
+    __tablename__ = 'sqlalchemy_documents'
     id = Column(Integer, primary_key=True)
     embedding = Column(VectorType(3))
 
-# or add hnsw index when creating table
-class TestWithIndex(Base):
-    __tablename__ = 'test_with_index'
-    id = Column(Integer, primary_key=True)
-    embedding = Column(VectorType(3), comment="hnsw(distance=l2)")
-
 Base.metadata.create_all(engine)
 ```
 
 Insert vector data
 
 ```python
-test = Test(embedding=[1, 2, 3])
-session.add(test)
+doc = Document(embedding=[1, 2, 3])
+session.add(doc)
 session.commit()
 ```
 
 Get the nearest neighbors
 
 ```python
-session.scalars(select(Test).order_by(Test.embedding.l2_distance([1, 2, 3.1])).limit(5))
+session.scalars(select(Document).order_by(Document.embedding.l2_distance([1, 2, 3.1])).limit(5))
 ```
 
 Get the distance
 
 ```python
-session.scalars(select(Test.embedding.l2_distance([1, 2, 3.1])))
+session.scalars(select(Document.embedding.l2_distance([1, 2, 3.1])))
 ```
 
 Get within a certain distance
 
 ```python
-session.scalars(select(Test).filter(Test.embedding.l2_distance([1, 2, 3.1]) < 0.2))
+session.scalars(select(Document).filter(Document.embedding.l2_distance([1, 2, 3.1]) < 0.2))
+```
+
+Add hnsw index
+
+```python
+# vector index currently depends on tiflash
+session.execute(text('ALTER TABLE sqlalchemy_documents SET TIFLASH REPLICA 1'))
+index = Index(
+    'idx_embedding',
+    func.vec_cosine_distance(Document.embedding),
+    mysql_prefix="vector",
+    mysql_using="hnsw"
+)
+index.create(engine)
```

Comment:
For other vector databases looks like the index is generally specified when table is created. I think this could be a better UX.

Reply:
sqlalchemy will create the table in two steps: 1. Creating a table without indexes. 2. Creating indexes. Since we need to explicitly set the tiflash replica when creating vector index separately, we cannot define the index in the table class.

Reply:
@wd0517 Thanks for the explain. So it is not possible to execute multiple statements there right? Let's raise some discussions with the PM to check how this can be workarounded.